fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/provider/impl/FileProvider.java

```diff
@@ -79,43 +83,104 @@ public long getSize(String workItem) throws FHIRException {
     public void readResources(long numOfLinesToSkip, String workItem) throws FHIRException {
         resources = new ArrayList<>();
         try {
-            long line = 0;
-            try {
-                if (br == null) {
-                    br = Files.newBufferedReader(Paths.get(getFilePath(workItem)));
-                }
-                // Imports must only skip lines if the size is higher, not equal to
-                // otherwise we start skipping the last line to skip.
-                for (int i = 0; i < numOfLinesToSkip; i++) {
-                    line++;
-                    br.readLine(); // We know the file has at least this number.
-                }
+            try (RandomAccessFile raf = new RandomAccessFile(Paths.get(getFilePath(workItem)).toFile(), "r")) {
```

Comment:
Random access might cause trouble depending on how the file-system is provided.

Reply:
Yes, that's right, it does speed up the import in this case. The downside of it is actually the same for us.  The prior code used read-to-line which is inefficient.  Some file systems use this approach of scan to location.  I think it's a nominal difference, and in most of the FileProvider usecases it's using the local file system on a developer machine.  

Let me know if there further downstream actions/preferences.

Reply:
I added BufferedInputStream inbetween the Inputstream and Count